sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/statefulOperators.scala

```diff
@@ -373,8 +389,21 @@ trait StateStoreWriter
     val storeMetrics = store.metrics
     longMetric("numTotalStateRows") += storeMetrics.numKeys
     longMetric("stateMemory") += storeMetrics.memoryUsedBytes
-    storeMetrics.customMetrics.foreach { case (metric, value) =>
-      longMetric(metric.name) += value
+    storeMetrics.customMetrics.foreach {
+      // Set the max for instance metrics
+      case (metric: StateStoreInstanceMetric, value) =>
```

Comment:
this code is a duplicate of the same code in `StreamingSymmetricHashJoinExec`. Should we just have a common func for this logic, instead of duplicating?

Reply:
Good idea, I've moved this logic to `setStoreCustomMetrics`